http/src/v0/refs.rs

```diff
@@ -253,11 +250,11 @@ fn iplds_refs<T: IpfsTypes>(
 
             if traverse_links {
                 for (link_name, next_cid) in ipld_links(&cid, ipld) {
-                    if unique && visited.contains(&next_cid) {
-                        // skip adding already yielded documents
+                    if unique && !queued_or_visited.insert(cid.clone()) {
+                        trace!("skipping already queued {}", cid);
```

Comment:
Clearly it should be `next_cid` instead of `cid` but testing locally first.

Reply:
Yep

Reply:
oof, didn't notice that it had changed